# FrozenLake
Read more at:

https://medium.com/swlh/frozen-lake-as-a-markov-decision-process-1692815ecfd1


https://zoo.cs.yale.edu/classes/cs470/materials/hws/hw7/FrozenLake.html

Documentation:
https://gymnasium.farama.org/environments/toy_text/frozen_lake/

In [5]:
!pip install gymnasium

     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 953.9/953.9 kB 6.7 MB/s eta 0:00:00


In [6]:
# frozen-lake-ex3.py
import gymnasium as gym
from gymnasium.envs.toy_text.frozen_lake import generate_random_map

# Actions in human language
actions = {
    'Left': 0,
    'Down': 1,
    'Right': 2,
    'Up': 3
}

In [7]:
# Variabler
slippery=False
my_policy = ['Right'] + ['Right'] + ['Down'] + ['Down'] + ['Down'] + ['Right']

In [8]:
def my_frozenlake(slippery, my_policy):

    print("--- Super amazing policy that will win! ---")
    print(my_policy)
    print("Maze is slippery: ", slippery)

    env = gym.make("FrozenLake-v1", is_slippery=slippery, render_mode="human")
    env.reset()
    env.render()

    for a in my_policy:
        new_state, reward, done, truncated, info = env.step(actions[a])
        print()
        env.render()
        print("Reward: {:.2f}".format(reward))
        print("Action:",a)
        print(info)
        print("Position:",new_state)
        if done:
            if reward == 1:
                print("****You reached the goal!****")
            else:
                print("****You fell through a hole!****")
            break
    env.close()

In [9]:
# Kör functionen!
my_frozenlake(slippery, my_policy)

--- Super amazing policy that will win! ---
['Right', 'Right', 'Down', 'Down', 'Down', 'Right']
Maze is slippery:  False

Reward: 0.00
Action: Right
{'prob': 1.0}
Position: 1

Reward: 0.00
Action: Right
{'prob': 1.0}
Position: 2

Reward: 0.00
Action: Down
{'prob': 1.0}
Position: 6

Reward: 0.00
Action: Down
{'prob': 1.0}
Position: 10

Reward: 0.00
Action: Down
{'prob': 1.0}
Position: 14

Reward: 1.00
Action: Right
{'prob': 1.0}
Position: 15
****You reached the goal!****


In [ ]:
my_frozenlake(True, my_policy)

--- Super amazing policy that will win! ---
['Right', 'Right', 'Down', 'Down', 'Down', 'Right']
Maze is slippery:  True

Reward: 0.00
Action: Right
{'prob': 0.3333333333333333}
Position: 0

Reward: 0.00
Action: Right
{'prob': 0.3333333333333333}
Position: 1

Reward: 0.00
Action: Down
{'prob': 0.3333333333333333}
Position: 5
****You fell through a hole!****


Det alldra mest själklara är att man föll genom ett hål. Pobability sjunker till ~33%. Om man tänker på det helt logiskt att mazen har hål i sig och det är halt, snubben springer omkring i mazen och var han skulle villa svänga så glider han vidare in i ett hål